lib/koa-authenticator.js

```diff
@@ -13,11 +14,15 @@ module.exports = function(escherConfig = {}, logger) {
         yield next;
       }
     } catch (ex) {
-      if (logger && logger.error) {
-        logger.error('authentication_request_error', ex.message, ex);
+      if (ex instanceof AuthenticationError) {
+        if (logger && logger.error) {
+          logger.error('authentication_request_error', ex.message, ex);
+        }
+
+        this.throw(401, ex.message);
       }
 
-      this.throw(401, ex.message);
+      this.throw(ex);
```

Comment:
Also, this would result in an 500 error with the current error handler middleware in the boar-server, as that looks for err.code === 401 and this.throw will create an error with err.status === 401. Note the difference between .code vs .status